packages/syft/src/syft/service/user/user_service.py

```diff
@@ -84,6 +91,212 @@ def create(
+        if user_role == ServiceRole.ADMIN:
+            return SyftError(
+                message="You can't request password reset for an Admin user."
+            )
+
+        user.reset_token = self.generate_new_password_reset_token(
+            context.server.settings.pwd_token_config
+        )
+        user.reset_token_date = datetime.now()
+
+        result = self.stash.update(
+            credentials=context.credentials, user=user, has_permission=True
+        )
+        if result.is_err():
+            return SyftError(
+                message=(
+                    f"Failed to update user with UID: {uid}. Error: {str(result.err())}"
+                )
+            )
+
+        return user.reset_token
+
+    @service_method(
+        path="user.reset_password", name="reset_password", roles=GUEST_ROLE_LEVEL
+    )
+    def reset_password(
+        self, context: AuthedServiceContext, token: str, new_password: str
+    ) -> SyftSuccess | SyftError:
+        """Resets a certain user password using a temporary token."""
+        result = self.stash.get_by_reset_token(
+            credentials=context.credentials, token=token
+        )
+        invalid_token_error = SyftError(
+            message=("Failed to reset user password. Token is invalid or expired!")
+        )
+
+        if result.is_err():
+            return SyftError(message="Failed to reset user password.")
+
+        user = result.ok()
+
+        # If token isn't found
+        if user is None:
+            return invalid_token_error
+
+        now = datetime.now()
+        time_difference = now - user.reset_token_date
+
+        # If token expired
+        if time_difference > timedelta(minutes=30):
+            return invalid_token_error
+
+        if not validate_password(new_password):
+            return SyftError(
+                message="Your new password must have at least 8 \
+                characters, Upper case and lower case characters\
+                and at least one number."
+            )
+
+        salt, hashed = salt_and_hash_password(new_password, len(new_password))
+        user.hashed_password = hashed
+        user.salt = salt
+
+        user.reset_token = None
+        user.reset_token_date = None
+
+        result = self.stash.update(
+            credentials=context.credentials, user=user, has_permission=True
+        )
+        if result.is_err():
+            return SyftError(
+                message=(f"Failed to update user password.  Error: {str(result.err())}")
+            )
+        return SyftSuccess(message="User Password updated successfully!")
+
+    def generate_new_password_reset_token(
+        self, token_config: PwdTokenResetConfig
+    ) -> str:
+        valid_characters = ""
+        if token_config.ascii:
+            valid_characters += string.ascii_letters
+
+        if token_config.numbers:
+            valid_characters += string.digits
+
+        # valid_characters = string.ascii_letters + string.digits
```

Comment:
I assume this is leftover and can be removed?

Reply:
Yup! Removed at: [74f1b9d](https://github.com/OpenMined/PySyft/pull/9052/commits/74f1b9da8124b0468a2c59fca716d17e103f8916)